### ISSUE TITLE
Feature request: Timeout encryption password

### ISSUE THREAD

**Issue by redstreet** [gsantner/markor#1250]
[Encryption](https://github.com/gsantner/markor/pull/887) is an awesome start, thank you @opensource21 and @gsantner for getting this in. For me, this was a critical feature.

The existing feature addresses the data at rest encryption. This feature request is to add a feature to making "locking" easy. This allows one to let someone else use their phone for a few minutes, unsupervised, without fear that private notes can be read. This can be accomplished in one or more of these ways:
- by having Markor forget the password on a configurable timeout (eg: after 5mins)
- by having Markor forget the password when exiting the app
- by placing a "forget" button on the main app screen

Is one or more of the above an easy feature to implement? Happy to hash this out further, provide examples of other open source apps that do this, etc. Thanks much for considering!

May I also take a moment to say how awesome Markor is. It's exactly what I was looking for. Thank you for building, sharing, and maintaining it @gsantner!

**Comment by opensource21** [comment]
I think there are very simple solutions:

- Use the app protection from android to disable access to Markor
- Simply set the password to 123456 instead of delete the password.

**Comment by redstreet** [comment]
Of course, those are simple solutions....and also extremely inconvenient to the point of being unusable if done 20 times a day. I already tried :).

Automation (the ideas I suggested above) removes the friction and increases security. These methods are pretty standard in most apps that deal with security, hence the request. If it's not trivial to implement any of them, I understand. I was hoping "forget while exiting" would be the easiest and most trivial option to implement.

**Comment by gsantner** [comment]
I say, clearing / setting random password / whatever value is a easy - so a password input/wrong dialog triggers - is the easiest option. And some settings option, always ask for password when restarting the app (=onCreate at MainActivity).

**Comment by gsantner** [comment]
I do say `clearing / setting random password / whatever value` is a easy option.  
So a password input dialog triggers.

Plus some settings option, `always ask for password when restarting the app` (=onCreate at MainActivity).

**Comment by sk-Prime** [comment]
I am using markor since last week. It is a great app. Only thing I miss is the password protection.

**Comment by gsantner** [comment]
This issue is about consuming passwords "per-file", encryption itself is already there.

You can enable it in settings. One password for all.

**Comment by sk-Prime** [comment]
Yes there is an option for that, for example `test.md.jenc` is an encrypted file, anyone can open it from Markor, isn't it? So if anyone (e.g family members/kid) take my phone and they can see it. Am I missing something?
I thought that requiring a password to open encrypted file will be a good feature.

**Comment by gsantner** [comment]
what you seemingly look for is more, locking Markor so you cannot use it without password.

**Comment by sk-Prime** [comment]
Yes, you are right but not locking entire markor. Ask password only for encrypted file.
e.g. `test.md.jenc` is an encrypted file. So if i try to open it, there will be a popup dialog box which will ask for decrypt password. If i can not input correct password it will say password mismatch, otherwise open the file.

**Comment by st3ss** [comment]
@opensource21 Love markor and its simplicity, thanks! 
Sorry about the irrelevancy but, i'm an idiot and changed the password so now i don't have access to my files. Is it possible to fix that? I changed x20+ times the password with no luck, if that makes it worse

**Comment by opensource21** [comment]
If you forgot the old password  the content is lost. That's the price for good encryption.

**Comment by nutpantz** [comment]
How about file encryption on or off toggle under "file settings" 
If it's on then then you can read and write files encrypted with the master password in settings.
If it is off then nothing is encrypted or unencrypted 
If you are editing a existing file that was not encrypted and you turn encryption on it warns you before saving it.
If an encrypted file is being edited and encryption of turned off you are warned before it is saved.

I do like the encryption, but i don't need every file encrypted and i do need to share the files on a desktop.

**Comment by phil834a** [comment]
Hi, I support nutpantz suggestion.

**Comment by gsantner** [comment]
Merge requests are welcome!

**Comment by opensource21** [comment]
> I do like the encryption, but i don't need every file encrypted and i do need to share the files on a desktop.  

I don't understand your feature request. You can decide for each file at creation time what type of file it should be. Only md.jenc files are encrypted.

Furthermore you can encrypt the files at desktop.

**Comment by phil834a** [comment]
Hi, thank you very much !!! I missed this per file encryption feature, sorry about that (I am definitely a regular end users ;O) I should have tried. 
So when a general encryption password is set, there is a file encryption checkbox which names the file as xxx.md or xxx.md.jenc, it also works if we set the jenc extension manually.
....
When I read it outside Markor the file is readable but not understandable (sorry English is not my native language).
When I read it through Markor it is decrypted automatically...and I don't like that...
Protecting the Markor app is painful from my opinion.
What about the Aug 13 2021 request ?
" Yes, you are right but not locking entire markor. Ask password only for encrypted file.
e.g. test.md.jenc is an encrypted file. So if i try to open it, there will be a popup dialog box which will ask for decrypt password. If i can not input correct password it will say password mismatch, otherwise open the file. "

There is also an issue as the file encryption password parameter, while not readable, can be changed easily . I dialog box asking to enter the current password to allow for a modification would be a safe enhancement I think. (...yes I feel that protecting the  Markor app is painful). 

Thank's for your support.

**Comment by gsantner** [comment]
First of all, for all kinds of improvements, needs, askinfs and "what about"s - it needs somebody to do it - to work on that part of code.

**Comment by phil834a** [comment]
Hi, I understand, 100% with you ! 
and if this work relies on real world user requirements it is where it adds value ;O)
Within plenty of alternatives I use Markor because it corresponds to most of my needs. 1000 thank's for that and for your time.

**Comment by clach04** [comment]
> 
> And i reading that right? Any file that does not have the extension .md.jenc will not be encrypted? And any file that has the extension. .md.jenc will be encrypted?

That's my understanding.

> 
> Is that in the faq? I'm sure i read it a few times and never saw that.

I don't think so but you could open a PR to add an entry to https://github.com/gsantner/markor/blob/master/README.md#FAQ :-)

**Comment by arichiardi** [comment]
Was looking at Markor and another solution that would not interfere with the password mechanics is called in other apps an App PIN.
